backend/src/main/java/ch/puzzle/pcts/model/certificate/Certificate.java

```diff
@@ -0,0 +1,194 @@
+
+import static org.apache.commons.lang3.StringUtils.trim;
+
+import ch.puzzle.pcts.model.Model;
+import ch.puzzle.pcts.model.certificatetype.CertificateType;
+import ch.puzzle.pcts.model.member.Member;
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.util.Objects;
+import org.hibernate.annotations.SQLDelete;
+
+@Entity
+@SQLDelete(sql = "UPDATE certificate SET deleted_at = CURRENT_TIMESTAMP WHERE id = ?")
+public class Certificate implements Model {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull(message = "{attribute.not.null}")
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "member_id")
+    private Member member;
+
+    @NotNull(message = "{attribute.not.null}")
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "certificate_type_id")
+    private CertificateType certificateType;
+
+    @NotNull(message = "{attribute.not.null}")
+    private LocalDate completedAt;
+
+    private LocalDate validUntil;
+
+    private String comment;
+
+    @Column(name = "deleted_at", insertable = false, updatable = false)
+    private Timestamp deletedAt;
+
+    private Certificate(Builder builder) {
+        this.id = builder.id;
+        this.member = builder.member;
+        this.certificateType = builder.certificateType;
+        this.completedAt = builder.completedAt;
+        this.validUntil = builder.validUntil;
+        this.comment = trim(builder.comment);
+        this.deletedAt = null;
+    }
+
+    public Certificate() {
+    }
+
+    @Override
+    public Long getId() {
+        return id;
+    }
+
+    @Override
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public Member getMember() {
+        return member;
+    }
+
+    public void setMember(Member member) {
+        this.member = member;
+    }
+
+    public CertificateType getCertificateType() {
+        return certificateType;
+    }
+
+    public void setCertificateType(CertificateType certificateType) {
+        this.certificateType = certificateType;
+    }
+
+    public LocalDate getCompletedAt() {
+        return completedAt;
+    }
+
+    public void setCompletedAt(LocalDate completedAt) {
+        this.completedAt = completedAt;
+    }
+
+    public LocalDate getValidUntil() {
+        return validUntil;
+    }
+
+    public void setValidUntil(LocalDate validUntil) {
+        this.validUntil = validUntil;
+    }
+
+    public String getComment() {
+        return comment;
+    }
+
+    public void setComment(String comment) {
+        this.comment = trim(comment);
+    }
+
+    public Timestamp getDeletedAt() {
+        return deletedAt;
+    }
+
+    public void setDeletedAt(Timestamp deletedAt) {
+        this.deletedAt = deletedAt;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof Certificate that))
+            return false;
```

Comment:
```suggestion
        if (!(o instanceof Certificate that)) {
            return false;
        }
```
This is a nit I know but always make the `{ }` after an `if`. Sadly this cannot be checked by Spottless 😒